migrate.js

```diff
@@ -10,70 +10,157 @@ const WAIT_TIME = 10; // Wait time between iterations in seconds
 const DEFAULT_PAGE_SIZE = 64; // Page size for retrieving documents from the custom index
 
 async function runMigration(options) {
-  if (!options.source) throw new Error("Error running migration: 'source' option is required")
-  if (!options.target) throw new Error("Error running migration: 'target' option is required")
-  if (!options.collection) throw new Error("Error running migration: 'collection' option is required")
-  if (!options.timestamp) throw new Error("Error running migration: 'timestamp' option is required")
-  if (!options.endpoint) throw new Error("Error running migration: 'endpoint' option is required")
-  if (!options.parallelism) throw new Error("Error running migration: 'parallelism' option is required")
+  if (!options.source)
+    throw new Error("Error running migration: 'source' option is required");
+  if (!options.target)
+    throw new Error("Error running migration: 'target' option is required");
+  if (!options.timestamp)
+    throw new Error("Error running migration: 'timestamp' option is required");
+  if (!options.endpoint)
+    throw new Error("Error running migration: 'endpoint' option is required");
+  if (!options.parallelism)
+    throw new Error(
+      "Error running migration: 'parallelism' option is required",
+    );
 
   try {
+    const migrator = new MigrationClient({
+      sourceKey: options.source,
+      targetKey: options.target,
+      defaultPageSize: DEFAULT_PAGE_SIZE,
+      maxParallelism: options.parallelism,
+      endpoint: options.endpoint,
+    });
+
+    const collectionList = [];
+
+    if (!!options.collections && options.collections.length > 0) {
+      if (
+        !!options.indexes &&
+        options.collections.length !== options.indexes.length
+      ) {
+        throw new Error(
+          "Incorrect number of Indexes. The number of collections and indexes provided must be the same.",
+        );
+      }
+
+      const collectionNames = options.collections;
+
+      for (let i = 0; i < collectionNames.length; i++) {
+        const collectionName = collectionNames[i];
+        const indexName = !!options.indexes
+          ? options.indexes[i]
+          : "_migration_index_for_" + collectionName;
+
+        collectionList.push({
+          collectionName,
+          indexName,
+        });
+      }
+    } else {
+      if (options.index) {
+        throw new Error(
+          "Custom indexes specified, but cannot be used when running with multiple collections.",
+        );
+      }
+      console.log(
+        "No collection specified. Gathering a list of all Collections in the source DB...",
```

Comment:
I feel like maybe this should be a separate CLI option, `--migrate-all` or something.  That or `No collection specified, preparing to migrate ALL collections in ${database}.  Are you sure?  (y/N)`